### ISSUE TITLE
Type instantiation is excessively deep and possibly infinite

### ISSUE THREAD

**Issue by moonrailgun** [ecyrbe/zodios-react#473]
![image](https://github.com/ecyrbe/zodios-react/assets/6964737/0b924597-b38f-4319-8efd-4287c42a6f00)

typescipt always report error special in this scene

**Comment by soulchild** [comment]
@moonrailgun Try using `getKeyByPath()` instead. This solved the problem for us, see https://github.com/ecyrbe/zodios/issues/505#issuecomment-1683708771.

Unfortunately, Zodios starts to feel like abandonware to me, because there haven't been any updates for over half a year now. It's in a good state, but I feel like there's something inherently wrong with the type inference leading to (sometimes only sporadic) type errors. IMHO, this makes this otherwise awesome library feel kind of … brittle, I guess? 

And I'd really like to finally update all my Zodios projects to react-query@v5. @ecyrbe [promised](https://x.com/ecyrbedev/status/1755153829794824643?s=20) to make this happen, but I'm starting to worry we're not gonna see any updates… 😢

**Comment by moonrailgun** [comment]
oh, thanks for your reply, @soulchild . i never use getKeyByPath because i wanna make unify with my source code.

i think zodios's infer chain is too big and dont make typescript happy. after i use it after with 100+ endpoint....

and i found i still using `react-query@4.x`. its why i dont feel it has problem. Is any epic change in `react-query@5.x` which let you wanna use it?

**Comment by soulchild** [comment]
> and i found i still using react-query@4.x. its why i dont feel it has problem. Is any epic change in react-query@5.x which let you wanna use it?

Nah, not really. 😉 But I like having my dependencies as up to date as possible and not leave them falling behind.

I'm more worried that there haven't been ANY updates to the Zodios ecosystem in quite a long time.
